src/lib/y2storage/disk_analyzer.rb

```diff
@@ -223,120 +173,97 @@ def mbr_gap
     # @see #scope
     #
     # @return [Hash{String => Array<Partition>}] see {#partitions_with_id}
-    def windows_partitions
-      @windows_partitions ||= begin
-        result = find_windows_partitions
-        log.info("Found Windows partitions: #{result}")
-        result
-      end
+    def windows_partitions(*disks, limit: DEFAULT_DISKS_LIMIT)
```

Comment:
The API can be somehow weird
```
all_disks.size #=> 15
analyzer.windows_partitions #=> Only analyzes 10, that's fine
analyzer.windows_partitions(all_disks) #=> Only analyzes 10 although I'm explicitly passing 15
analyzer.windows_partitions(all_disks, limit: nil) #=> This is the correct way to analyze them all
```
Should we maybe (just asking, not saying) honor the `limit` only if `*disks` is empty?
```
all_disks.size #=> 15
analyzer.windows_partitions #=> Only analyzes 10, that's fine
analyzer.windows_partitions(all_disks) #=> Analyzes all the 15 passed ones
analyzer.windows_partitions(all_disks, limit: 10) #=> Only analyzes 10, although it made me pass the 10 (default) explicitly
```
Both options are weird at some point. I'm just not sure what is less surprising.